controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -304,31 +333,45 @@ private boolean acceptNewApplicationVersionNow(LockedApplication application) {
         return false;
     }
 
+    private Optional<BuildService.BuildJob> toBuildJob(ApplicationId applicationId, JobType jobType) {
+        return applications().get(applicationId)
+                             .flatMap(application -> application.deploymentJobs().projectId())
+                             .map(projectId -> new BuildService.BuildJob(projectId, jobType.jobName()));
+    }
+
+
     public static class Triggering {
 
-        private final LockedApplication application; // TODO jvenstad: Consider passing an ID instead.
+        private final ApplicationId id;
         private final JobType jobType;
-        private final boolean retry;
+        private final long projectId;
+        private final Change change;
         private final String reason;
+        private final Instant availableSince;
```

Comment:
I'll keep the current name, since it's not the time of creation of the `Job`, but rather the time at which the change became available for the step the `Job` belongs to.